React/Modules/RCTUIManager.m

```diff
@@ -1053,6 +1059,16 @@ - (void)_manageChildren:(NSNumber *)containerTag
   }];
 }
 
+- (void)synchronouslyUpdateViewOnUIThread:(NSNumber *)reactTag
+                                 viewName:(NSString *)viewName
+                                    props:(NSDictionary *)props
+{
+  RCTAssertMainQueue();
+  RCTComponentData *componentData = _componentDataByName[viewName];
+  UIView *view = _viewRegistry[reactTag];
+  [componentData setProps:props forView:view];
```

Comment:
Hmm, in what scenario would this be true? No prop should be driven both by the shadow view and a native animation, it's one or the other. Native animated props actually get stripped from the set of props sent to the shadow view; they are mutated independently via this package, completely outside the shadow view hierarchy. It's a limitation for sure, e.g. we can't drive layout props like width or height, but this should be safe from a race.